core/market/decentralized/src/rest_api/common.rs

```diff
@@ -6,18 +6,35 @@ use ya_service_api_web::middleware::Identity;
 use ya_std_utils::ResultExt;
 
 use super::PathAgreement;
+use crate::db::model::OwnerType;
 use crate::market::MarketService;
+use crate::negotiation::error::AgreementError;
 
 #[actix_web::get("/agreements/{agreement_id}")]
 async fn get_agreement(
     market: Data<Arc<MarketService>>,
     body: Path<PathAgreement>,
     id: Identity,
 ) -> impl Responder {
-    let agreement_id = body.into_inner().agreement_id;
-    market
-        .get_agreement(&agreement_id, &id)
-        .await
-        .inspect_err(|e| log::error!("[GetAgreement] {}", e))
-        .map(|agreement| HttpResponse::Ok().json(agreement))
+    // We don't know, if we are requestor or provider. Try to get Agreement for both sides
```

Comment:
we should know. We have identity

Reply:
You mean `Identity.role`? Or how you determine, who we are?